pkg/testutil/gcs.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/storage"
+)
+
+// UploadGCSFiles uploads an object to GCS bucket.
+func UploadGCSFile(ctx context.Context, gcsClient *storage.Client, bucket, object string, data io.Reader, metadata map[string]string) (func() error, error) {
+	closer := func() error {
```

Comment:
It is weird that the function called `UploadGCSFile` and it returns a closer function which delete the gcs file.

Reply:
Updated the naming and comment for this function.

Reply:
There is a bug as well - if the we failed to create the object, then deleting the object will fail as well. I agree with @capri-xiyue we can skip the cleanup in code and rely on GCS lifecycle to delete them.

Reply:
As I've mentioned in another comment, Just blindly return closer as non-nil doesn't fix the issue. For example, if you fail to create gcs object and then you return closer as non-nil, when you call closer(), it will throw error which brings noise. I think a better approach is to make this function just upload file to gcs and you can just do clean up separately in the e2e test instead of returning a closer here.

Reply:
Removed the closer function I will set up the gcs lifecycle to delete the files for e2e test when we set up that for the prober files.